pkg/controller/installlogmonitor/installlogmonitor_controller.go

```diff
@@ -0,0 +1,280 @@
+		iLog.WithError(err).Error("cannot get clusterdeprovisionrequest")
+		return reconcile.Result{}, err
+	}
+
+	if !cm.DeletionTimestamp.IsZero() {
+		iLog.Debug("configmap being deleted, skipping")
+		return reconcile.Result{}, nil
+	}
+
+	// Temporary migration of install logs created by hiveutil images prior to the labels being added.
+	// TODO: Remove this and watch for ConfigMaps with our labels once the main ClusterImageSets in use are
+	// new enough to populate the Hive labels on install log ConfigMaps.
+	isInstallLog, needsMigration := r.isHiveInstallLog(cm)
+	if !isInstallLog {
+		return reconcile.Result{}, nil
+	}
+
+	if needsMigration {
+		return reconcile.Result{}, r.migrateInstallLog(cm)
+	}
+
+	cd := &hivev1.ClusterDeployment{}
+	cdName, ok := cm.Labels[hivev1.HiveClusterDeploymentNameLabel]
+	if !ok {
+		iLog.Error("cannot reconcile install log configmap without cluster-deployment-name label")
+		return reconcile.Result{}, fmt.Errorf("install log configmap %s has no %s label",
+			request.NamespacedName.String(),
+			hivev1.HiveClusterDeploymentNameLabel)
+	}
+	cdNSName := types.NamespacedName{Namespace: cm.Namespace, Name: cdName}
+	err = r.Get(context.TODO(), cdNSName, cd)
+	if err != nil {
+		iLog.WithError(err).Error("unable to lookup cluster deployment for configmap")
+		return reconcile.Result{}, err
+	}
+
+	// Check if we've processed this configmap before and skip if so. The install manager currently
+	// replaces the log on a retry (full delete + recreate).
+	if _, ok := cm.Annotations[processedAnnotation]; ok {
+		iLog.Debugf("skipping install log that has already been processed (%s annotation present)", processedAnnotation)
+		return reconcile.Result{}, nil
+	}
+
+	iLog.Info("processing new install log")
+
+	// Apply the processed annotation and save before we report anything. If we succeed here and fail later
+	// we accept that we will lose processing of that install log. This is for tracking/metrics and not
+	// anything critical.
+	if cm.ObjectMeta.Annotations == nil {
+		cm.ObjectMeta.Annotations = map[string]string{}
+	}
+	cm.ObjectMeta.Annotations[processedAnnotation] = "true"
+	if err := r.Client.Update(context.Background(), cm); err != nil {
+		iLog.WithError(err).Error("error saving processed annotation")
+		return reconcile.Result{}, err
+	}
+
+	success, successKeyReported := cm.Data["success"]
+
+	foundError := false
+	if !successKeyReported || success == "false" {
+		log := cm.Data["log"]
+		// Log each line separately, this brings all our install logs from many namespaces into
+		// the main hive log where we can aggregate search results.
+		installLogLines := strings.Split(log, "\n")
+		for _, l := range installLogLines {
+			iLog.WithField("line", l).Info("install log line")
+		}
+
+		// Scan log contents for known errors if the install was not reported as a success:
+		for _, ilr := range ilRegexes {
+			for _, re := range ilr.SearchRegexes {
+				if re.Match([]byte(log)) {
```

Comment:
should the match be done per line (above)?

Reply:
I can't think of a reason, we just want to see if the regex matches the overall search string, I'm assuming that your regex wouldn't match across newlines unless you made it. Anything I'm missing though?

Reply:
Not with the regexes that you have now, but I was wondering whether at some point a regex could get a false positive because when you created it you were assuming it would be matched against a single line

Reply:
But yes, sorry you're right, unless you explicitly make it match across multiple lines, it won't ... so please ignore.